src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -1,46 +1,100 @@
+    Transformation section of the PHDI VA Pilot Architecture diagram. If
+    the message is successfully converted to FHIR, then it is processed
+    through the remaining transformations (standardization & geocoding),
+    and stored in both a storage container and the FHIR server. If it is
+    not valid, it is routed to a storage container marked as invalid, and
+    no further processing is done.
+    """
     salt = get_required_config("HASH_SALT")
     geocoder = get_smartystreets_client(
         get_required_config("SMARTYSTREETS_AUTH_ID"),
         get_required_config("SMARTYSTREETS_AUTH_TOKEN"),
     )
-    fhirserver_cred_manager = get_fhirserver_cred_manager(
-        get_required_config("FHIR_URL")
-    )
-
     container_url = get_required_config("INTAKE_CONTAINER_URL")
-    container_prefix = get_required_config("INTAKE_CONTAINER_PREFIX")
-    output_path = get_required_config("OUTPUT_CONTAINER_PATH")
+    valid_output_path = get_required_config("VALID_OUTPUT_CONTAINER_PATH")
+    invalid_output_path = get_required_config("INVALID_OUTPUT_CONTAINER_PATH")
+
+    bundle = convert_message_to_fhir(
+        message=message,
+        input_data_type=message_mappings["input_data_type"],
+        root_template=message_mappings["root_template"],
+        template_collection=message_mappings["template_collection"],
+        access_token=access_token,
+        fhir_url=fhir_url,
+    )
 
-    for datatype, bundle in read_fhir_bundles(container_url, container_prefix):
+    if bundle:
         transform_bundle(geocoder, bundle)
         add_patient_identifier(salt, bundle)
-        store_bundle(container_url, output_path, bundle, datatype)
-        upload_bundle_to_fhir_server(fhirserver_cred_manager, bundle)
+        store_data(
+            container_url,
+            valid_output_path,
+            f"{message_mappings['filename']}.fhir",
+            message_mappings["bundle_type"],
+            bundle=bundle,
+        )
+        upload_bundle_to_fhir_server(cred_manager, bundle)
+    else:
+        store_data(
+            container_url,
+            invalid_output_path,
+            f"{message_mappings['filename']}.hl7",
+            message_mappings["bundle_type"],
+            message=message,
+        )
 
 
-def main(req: func.HttpRequest) -> func.HttpResponse:
+def main(blob: func.InputStream) -> func.HttpResponse:
     logging.info("Triggering intake pipeline")
+
+    fhir_url = get_required_config("FHIR_URL")
+    cred_manager = get_fhirserver_cred_manager(fhir_url)
+
     try:
-        run_pipeline()
+        access_token = cred_manager.get_access_token()
+
+        # VA sends \\u000b & \\u001c in real data, ignore for now
+        messages = convert_batch_messages_to_list(
+            blob.read().decode("utf-8", errors="ignore")
+        )
+        message_mappings = get_file_type_mappings(blob.name)
+
+        for i, message in enumerate(messages):
+            message_mappings["filename"] = generate_filename(blob.name, i)
```

Comment:
The way I thought about to solve this would be to move `get_file_type_mappings` into the `for` loop and generate the mappings for each message. This would be necessary because we need the message index so we know which message it maps back to in the parent file. Generating the mappings for each message felt like overkill, so this is a small price to pay to reduce the number of times the function gets called.

Reply:
I see both points, and probably too small a difference to worry much about.  Perhaps a middle ground would be to split the parameter into `file_type_mappings` and `message_mappings`.  The former are things that could be derived from the file, and the latter could be message-specific info (like output filename).